src/ini.pest

```diff
@@ -1,13 +1,17 @@
-WHITESPACE = _{ " " }
+ini = { SOI ~ (section | pair | comment | blank_line)* ~ ws? ~ EOI }
+ws = { (" " | "\t")* }
+
+// section
+section = { ws ~ "[" ~ name ~ "]" ~ ws ~ newline? }
 char = { ASCII_ALPHANUMERIC | "." | "_" | "/" }
+newline = { NEWLINE }
+
+// key-value pair
+pair = { ws ~ name ~ ws ~ equals ~ ws ~ value ~ ws ~ newline? }
+equals = { "=" }
```

Comment:
I don't love that I had to do this, but I couldn't think of a better way to distinguish from whitespace after the key vs. whitespace before the value.

https://pest.rs/?bin=14txp2#editor

Would love to see some solution that involved slicing.

Reply:
Yeah - explicit whitespace-handling really distracts me. I would really appreciate, if we would find a more "elegant" solution.